Signal/src/AppDelegate.m

```diff
@@ -49,22 +49,33 @@ @implementation AppDelegate
 - (void)performUpdateCheck{
     NSString *previousVersion = Environment.preferences.lastRanVersion;
     NSString *currentVersion = [Environment.preferences setAndGetCurrentVersion];
+    // TODO: remove
     if (!previousVersion) {
         DDLogError(@"No previous version found. Possibly first launch since install.");
         [Environment resetAppData]; // We clean previous keychain entries in case their are some entries remaining.
     }
-    else if ([Environment.preferences getIsMigratingToVersion2Dot0] || [currentVersion compare:previousVersion options:NSNumericSearch] == NSOrderedDescending){
-        if([self isVersion:previousVersion atLeast:@"1.0.2"]) {
-            [VersionMigrations migrateFrom1Dot0Dot2ToVersion2Dot0]; // this is only necessary for older apps
+    else if(([self isVersion:previousVersion atLeast:@"1.0.2" andLessThan:@"2.0"]) || [Environment.preferences getIsMigratingToVersion2Dot0] ) {
```

Comment:
getIsMigratingToVersion2Dot0 is stored in preferences not DB as we may not in general have a DB already if the migration was halted in the midst